### PR TITLE
ROC-7247: Case level experts

### PR DESCRIPTION
**Pull request by daveoh**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-7247

### Change description ###
Enable case level experts

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java [hunk 1]
```diff
@@ -48,13 +48,9 @@ public void from(CCDCase ccdCase, Claim.ClaimBuilder claimBuilder) {
...
 
-        Optional.ofNullable(ccdCase.getExpertReportPermissionPartyGivenToClaimant())
+        Optional.ofNullable(ccdCase.getGrantExpertReportPermission())
             .ifPresent(permission -> builder
-                .expertReportPermissionGivenToClaimant(YesNoOption.valueOf(permission.name())));
-
-        Optional.ofNullable(ccdCase.getExpertReportPermissionPartyGivenToDefendant())
-            .ifPresent(permission -> builder
-                .expertReportPermissionGivenToDefendant(YesNoOption.valueOf(permission.name())));
+                .grantExpertReportPermission(YesNoOption.valueOf(permission.name())));
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java]: Is there a distinction between `NO` and `null` here?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java [hunk 2]
```diff
@@ -90,6 +91,7 @@ public CallbackResponse prepopulateOrder(CallbackParams callbackParams) {
         data.put(DOC_UPLOAD_FOR_PARTY, BOTH.name());
         data.put(EYEWITNESS_UPLOAD_FOR_PARTY, BOTH.name());
         data.put(PAPER_DETERMINATION, NO.name());
+        data.put(GRANT_EXPERT_REPORT_PERMISSION, NO);
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java]: Not `NO.name()`?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java [hunk 3]
```diff
@@ -124,7 +126,7 @@ private void addCourtData(Claim claim, CCDCase ccdCase, Map<String, Object> data
         CCDRespondent respondent = ccdCase.getRespondents().get(0).getValue();
 
         CCDResponseRejection claimantResponse = Optional.ofNullable(respondent.getClaimantResponse())
-            .map(r -> (CCDResponseRejection) r)
+            .map(CCDResponseRejection.class::cast)
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java]: 👍

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java (hunk 2 above)
Comment (daveoh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java]: It's an enum being cast to a string so name() is not needed. Was in 2 minds about cleaning all the other calls up

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java (hunk 1 above)
Comment (daveoh) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java]: No. We are only interested in an explicit yes.

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/HearingCourtDetailsFinderTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/GenerateOrderRule.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4936730&bid=16324767)

**Pull request merged**